src/HotChocolate/Core/test/Validation.Tests/FieldsOnCorrectTypeRule.cs

```diff
@@ -0,0 +1,305 @@
+﻿using System.Linq;
+using HotChocolate.Language;
+using Microsoft.Extensions.DependencyInjection;
+using Snapshooter.Xunit;
+using Xunit;
+
+namespace HotChocolate.Validation
+{
+    public class FieldsOnCorrectTypeRuleTests
+        : DocumentValidatorVisitorTestBase
+    {
+        public FieldsOnCorrectTypeRuleTests()
+            : base(services => services.AddFieldRules())
+        {
+        }
+
+        [Fact]
+        public void GoodObjectFieldSelection()
+        {
+            ExpectValid(@"
+                fragment objectFieldSelection on Dog {
+                    __typename
+                    name
+                }
+                
+                query {
+                    dog {
+                         ...objectFieldSelection
+                    } 
+                }
+            ");
+        }
+
+        [Fact]
+        public void GoodAliasedObjectFieldSelection()
```

Comment:
This one will fail I think. So what they are testing here is that  the validation skips field validation on unknown types

Reply:
This might be specific to their implementation.